File: golang/examples/basic/consumer.go
package main

import (
	"log"
	"os"
	"os/signal"
	"syscall"

	"github.com/confluentinc/confluent-kafka-go/v2/kafka"
	kafkalib "github.com/kafka-examples/golang/src/kafka"
)

func main() {
	// Создаем логгер
	logger := log.New(os.Stdout, "basic-consumer: ", log.LstdFlags)
	logger.Println("Запуск консьюмера...")

	// Название топика
	topics := []string{"basic-topic"}

	// Создаем конфигурацию
	config := map[string]string{
		"group.id":          "basic-consumer-group",
		"auto.offset.reset": "earliest", // Начинаем с самого раннего сообщения
	}

	// Создаем консьюмера
	consumer, err := kafkalib.NewConsumer(topics, config, logger)
	if err != nil {
		logger.Fatalf("Ошибка при создании консьюмера: %v", err)
	}
	defer consumer.Close()

	// Обработчик CTRL+C для грациозного завершения
	sigchan := make(chan os.Signal, 1)
	signal.Notify(sigchan, syscall.SIGINT, syscall.SIGTERM)

	// Запускаем обработку в отдельной горутине
	go func() {
		<-sigchan
		logger.Println("Получен сигнал завершения")
		consumer.Stop()
	}()

	// Создаем обработчик сообщений
	messageHandler := func(message *kafka.Message) bool {
		logger.Printf("Получено сообщение: %s", string(message.Value))

		// Для демонстрации, можем проверить содержимое сообщения
		// и остановить обработку по определенному условию
		// if strings.Contains(string(message.Value), "stop") {
		//     return false  // остановить обработку
		// }
		
		return true // продолжать обработку
	}

	// Запускаем консьюмера
	consumer.Start(messageHandler)

	logger.Println("Консьюмер остановлен")
}

File: php/examples/basic/consumer.php
<?php

require_once __DIR__ . '/../../vendor/autoload.php';

use Monolog\Logger;
use Monolog\Handler\StreamHandler;

// Создаем логгер
$logger = new Logger('kafka-consumer');
$logger->pushHandler(new StreamHandler('php://stdout', Logger::DEBUG));

// Название топика
$topic = 'test-topic';

// Конфигурация консьюмера
$conf = new RdKafka\Conf();
$conf->set('metadata.broker.list', 'kafka:29092');
$conf->set('group.id', 'php-consumer-group');
$conf->set('auto.offset.reset', 'earliest');

// В новых версиях rdkafka нельзя передавать замыкания напрямую в метод set()
$logger->info("Настраиваем конфигурацию Kafka");

// Создаем консьюмера
$consumer = new RdKafka\KafkaConsumer($conf);

// Подписываемся на топик
$consumer->subscribe([$topic]);

$logger->info("Начинаем слушать топик {$topic}");
$logger->info("Для выхода нажмите Ctrl+C");



// Читаем сообщения
while (true) {
    $message = $consumer->consume(10000);
    
    switch ($message->err) {
        case RD_KAFKA_RESP_ERR_NO_ERROR:
            $logger->info("Получено сообщение: {$message->payload}");
            $logger->debug("Топик: {$message->topic_name}, Раздел: {$message->partition}, Смещение: {$message->offset}, Ключ: " . ($message->key ?: 'NULL'));
            break;
            
        case RD_KAFKA_RESP_ERR__PARTITION_EOF:
            $logger->info("Достигнут конец раздела");
            break;
            
        case RD_KAFKA_RESP_ERR__TIMED_OUT:
            $logger->debug("Тайм-аут, продолжаем...");
            break;
        default:
            $logger->error("Ошибка: {$message->errstr()}");
            break;

    }
}
